src/services/users/get-invited-user-detail.test.js

```diff
@@ -0,0 +1,61 @@
+const { expect } = require('chai')
+const { faker } = require('@faker-js/faker')
+const db = require('../../db')
+const {fixtures} = require('../../../test/utils')
+const getInvitedUserDetail = require('./get-invited-user-detail')
+
+
+describe('GET invited-user information', () => {
+	
+	let invalidInviteToken
+
+	before(async () => {
+		
+		invalidInviteToken = fixtures.generateAccessToken({
+			email : faker.internet.email()
+		})
+			
+	})
+
+	describe('Failure', async()=>{
+
+		it('should throw an error if the invite is invalid', async () =>
+			expect(getInvitedUserDetail(invalidInviteToken))
+				.to.be.rejectedWith(
+					'Invite is invalid/expired')
+		)
+	})
+
+	describe('Success', async()=>{
+
+		let inviteUserData
+		let inviteToken
+
+		before(async () => {
+			inviteUserData = {
+				email : faker.internet.email()
+			}
+
+			inviteToken = fixtures.generateAccessToken(inviteUserData)
+			await fixtures.insertUserInvite(
+				{email: inviteUserData.email})
+		
+		})
+		it('should get invited user information', async () =>{
+			const {email} = inviteUserData
+			const query = await getInvitedUserDetail(inviteToken)
+			const {rows} = await db.query(
+				`SELECT * FROM user_invites
+			 WHERE email = $1`,[email] )
+			const result = rows[0]
+		
+			return expect(result).to.eql({
```

Comment:
so our concern is mostly about what the service function is returning so that if it changes for some reason, we will be able to catch it

So it should us comparing `invitedUser`  we got to any other data

also no need to make a query, i guess the `await fixtures.insertUserInvite(
				{email: inviteUserData.email})` function in the `before` should return the invited user data as well after the insert

Reply:
sure